test/base-tlsutility.cpp

```diff
@@ -2,11 +2,61 @@
 
 #include "base/tlsutility.hpp"
 #include <BoostTestTargetConfig.h>
+#include <functional>
+#include <memory>
+#include <openssl/asn1.h>
+#include <openssl/bn.h>
+#include <openssl/evp.h>
+#include <openssl/obj_mac.h>
+#include <openssl/rsa.h>
+#include <openssl/x509.h>
 #include <utility>
 #include <vector>
 
 using namespace icinga;
 
+static EVP_PKEY* GenKeypair()
+{
+	InitializeOpenSSL();
+
+	auto e (BN_new());
+	BOOST_REQUIRE(e);
+
+	auto rsa (RSA_new());
+	BOOST_REQUIRE(rsa);
+
+	auto key (EVP_PKEY_new());
+	BOOST_REQUIRE(key);
+
+	BOOST_REQUIRE(BN_set_word(e, RSA_F4));
+	BOOST_REQUIRE(RSA_generate_key_ex(rsa, 4096, e, nullptr));
+	BOOST_REQUIRE(EVP_PKEY_assign_RSA(key, rsa));
+
+	return key;
+}
+
+static std::shared_ptr<X509> MakeCert(char* issuer, EVP_PKEY* signer, char* subject, EVP_PKEY* pubkey, std::function<void(ASN1_TIME*, ASN1_TIME*)> setTimes)
```

Comment:
The `char *` parameters should be const, just noticed a new compiler warning scrolling by due to this:

```
.../test/base-tlsutility.cpp:92:36: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
   92 |  BOOST_CHECK(IsCaUptodate(MakeCert("Icinga CA", key, "Icinga CA", key, [](ASN1_TIME* notBefore, ASN1_TIME* notAfter) {
      |                                    ^~~~~~~~~~~
```

Reply:
But they're (unsigned char*)ed anyway.

Reply:
Gosh, I hate OpenSSL [1.0.2](https://www.openssl.org/docs/man1.0.2/man3/X509_NAME_add_entry_by_NID.html) (or those Linux distributions that think it's a good idea to keep that version alive), in [1.1.1](https://www.openssl.org/docs/man1.1.1/man3/X509_NAME_add_entry_by_NID.html) they realized that making that parameter `const` is probably a good idea.

Anyways, then please use `const_cast<>()`, that's more of a "I'm intentionally doing this" and shouldn't issue a warning.

Reply:
1. Don't complain about your life. When I started working here, until my final exam, we had to support RHEL 5. I.e. Python 2.4, C++03 + auto and OpenSSL 0.9.8e with TLSv1.0. Not to mention the PHP version.
2. The current cast does neither despite -Wall -Wextra (and is how our actual code does this).

Reply:
I mean we're writing software in C++, not in C, but I get it, for some reason, you and C-style casts are just inseparable.